### PR TITLE
WIP - Metadata

### PR DESCRIPTION
**Pull request by Rucknar**
**Changes:**
* Moved to sourcing information from metadata service
* No longer requires API Access Key + Secret
* Simplified codebase to match the simpler metadata service data
* Moved to constmetrics, dropped locking
* Moved to multi-stage builds
* Made use of go-common

**Still to do:**
* Add back in health status for services & stacks, requires changes to metadata service
* Debug logging, fix handling and passing around the go-common logging 
* Add back in state for hosts, requires changes to metadata service
* Improve how we handle the rancher-metadata URL and version
* Resolve 'built-in' metrics through the go-common router. Currently initialised but not appearing.
* If possible, obtain possible states from rancher directly or from a standard library

### CODE REVIEW

**Comment by Rucknar** [comment]
@flock3 Added you as a reviewer, not yet finished but would appreciate any feedback

File: collect.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+			continue
+		}
+
+		ch <- prometheus.MustNewConstMetric(e.rancherMetrics["servicesScale"], prometheus.GaugeValue, float64(x.Scale), x.Name, x.StackName)
+
+		eLogger.Debug("Processing State metrics")
+
+		for _, s := range e.ServiceStates {
+			if x.State == s {
+				ch <- prometheus.MustNewConstMetric(e.rancherMetrics["servicesState"], prometheus.GaugeValue, float64(1), x.Name, x.StackName, s)
```
Comment (flock3) [inline collect.go]: Could just change float64(1) and (0) to be a var defined

File: collect.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+		eLogger.Debug("Processing stack metrics")
+
+		// Checks the metric is of the expected kind
+		if x.Kind != "stack" {
+			eLogger.Errorf("Error processing stack metric, metric not of expected kind")
+			continue
+		}
+
+		eLogger.Warn("Processing State metrics not currently possible, awaiting feature request in metadata service")
```
Comment (flock3) [inline collect.go]: Is this a crit rather than a warn - you are literally saying "I can't do the thing you asked me to do"

File: request.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", url, nil)
+	req.Header.Set("Accept", "application/json")
+
+	resp, err := client.Do(req)
+
+	if err != nil {
+		eLogger.Error("Error Collecting JSON from API: ", err)
+		panic(err)
```
Comment (flock3) [inline request.go]: I wouldn't panic (as they're ungraceful to handle) but instead return a more manageable error

**Pull request closed without merging**